File: onecmonitor-server/Models/TechLogSettings.cs
using System.ComponentModel.DataAnnotations;

namespace OnecMonitor.Server.Models;

public class TechLogSettings : DatabaseObject
{
    public bool Enabled { get; set; }
    public Guid? DbmsId { get; set; }
    [MaxLength(100)]
    public string DatabaseName { get; set; } = string.Empty;
    [MaxLength(100)]
    public string Table { get; set; } = string.Empty;
    public Guid? CredentialsId { get; set; }
    
    public Dbms? Dbms { get; set; }
    public Credentials? Credentials { get; set; }
}

File: oneswiss-web/src/lib/api/techlog-settings.ts
import { apiGet, apiPut } from "@/lib/api/client";

export type TechLogLookupItem = {
  id: string;
  name: string;
};

export type TechLogSettingsItem = {
  enabled: boolean;
  dbmsId: string | null;
  databaseName: string;
  table: string;
  credentialsId: string | null;
};

export type TechLogSettingsResponse = {
  settings: TechLogSettingsItem;
  dbms: TechLogLookupItem[];
  credentials: TechLogLookupItem[];
};

export type SaveTechLogSettingsRequest = {
  enabled: boolean;
  dbmsId: string | null;
  databaseName: string;
  table: string;
  credentialsId: string | null;
};

export function getTechLogSettings() {
  return apiGet<TechLogSettingsResponse>("/api/techlog/settings");
}

export function saveTechLogSettings(request: SaveTechLogSettingsRequest) {
  return apiPut<TechLogSettingsItem, SaveTechLogSettingsRequest>("/api/techlog/settings", request);
}
